src/content/doc-sdk-golang/data-types.mdx

```diff
@@ -128,6 +128,46 @@ This SDK facilitates communication between client and the backend service using
 
 <br /><br />
 
+## Json encoding
+
+The SDK uses the `encoding/json` package to encode and decode JSON.
+
+```go
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/surrealdb/surrealdb.go/pkg/models"
+)
+
+type Person struct {
+	ID       *models.RecordID     `json:"id,omitempty"`
+	Name     string               `json:"name"`
+	Surname  string               `json:"surname"`
+	Location models.GeometryPoint `json:"location"`
+}
+
+type CustomRecordID struct {
+	models.RecordID
+}
+
+func (r CustomRecordID) MarshalJSON() ([]byte, error) {
+	return json.Marshal(fmt.Sprintf("%s:%v", r.Table, r.ID))
+}
+
+type PersonWithCustomID struct {
+	ID       CustomRecordID       `json:"id,omitempty"`
+	Name     string               `json:"name"`
+	Surname  string               `json:"surname"`
+	Location models.GeometryPoint `json:"location"`
+}
+
+```
+
+
+
```

Comment:
Probably this should be removed because we don't use `encoding/json` nor does it support the `MarshalJSON` API.
We use CBOR with options to choose the CBOR implementation from either "fxamacker/cbor"-based implementation or our latest `surrealcbor` implementation.

fxamacker/cbor-based: https://github.com/surrealdb/surrealdb.go/blob/main/pkg/models/cbor.go

surrealcbor: https://github.com/surrealdb/surrealdb.go/tree/main/surrealcbor